### PR TITLE
Percy Local Report Generation

### PR DESCRIPTION
**Pull request by princebaretto99**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +24,13 @@
     "percy-report": "./bin/cli.js"
   },
   "dependencies": {
+    "@percy/cli": "^1.10.3",
```
Comment (nithyamn) [inline package.json:27]: Can we upgrade to latest version?

File: src/template/report.html [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                <div class="col-md-4 text-center border p-2">
+                    <label>Snapshot Name</label>
+                    <select id="select-snapshot">
+                        <option selected>All</option>
+                        <% for(let snapshot of details){ %>
+                            <option value="<%= snapshot['name']  %>">
+                                <%= snapshot['name'] %>
+                            </option>
+                            <% } %>
+                    </select>
```
Comment (nithyamn) [inline src/template/report.html:158]: The UI breaks for longer snapshot names.

File: README.md [hunk 3]
```diff
@@ -1,25 +1,43 @@
...
 
 ```
 
 ## Usage
 
-```sh
+<h3>Using the Generate Report functionality independently</h3>
 
+```sh
 percy-report generate [options] <buildId>
```
Comment (nithyamn) [inline README.md:17]: Cosmetic change, lets keep <buildID> before [options]

File: src/template/report.html [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                let table = tables.item(i)
+                if(table.id !== "build-details"){
+                    table.hidden = !shouldEnable(table.dataset)
+                }
+            }
+        }
+    </script>
+</head>
+
+<body class="p-5">
```
[the remark this answers is not in the document]
Comment (Rushabhshroff) [reply src/template/report.html:98]: We'll implement this in future versions

File: README.md [hunk 5]
```diff
@@ -1,25 +1,68 @@
...
 
 ```
+export PERCY_TOKEN=<read-only | full-access>
+```
+
 
 ## Usage
 
-```sh
+<h3>Using the Generate Report functionality independently</h3>
```
Comment (nithyamn) [inline README.md:22]: Dont add "Using" to the sub-headings. Can be something like "Generate Report - Per Build"

File: README.md [hunk 6]
```diff
@@ -1,25 +1,68 @@
...
+Step 3 : Execute the Percy Report Generation Step<br>
+
+[Example](/example/percy.sh):
+```sh
+export PERCY_TOKEN=<your-percy-token>
+BUILD_ID=$(npx percy exec -- {Test Command} | grep build | awk -F "/" '{print $NF}')
+npx percy-report generate $BUILD_ID
+```
+
+<h3>Using Summary functionality </h3>
```
Comment (nithyamn) [inline README.md:53]: Dont add "Using" to the sub-headings. Can be something like "Generate Summary"

File: README.md [hunk 7]
```diff
@@ -1,25 +1,68 @@
...
+percy-report [options] summary <project-slug>
+```
+
+Note: Project Slug is equal to Project Name on your percy dashboard
+
+<h3>Options for summary functionlity</h3>
+
+```sh
+  --percy-token <percyToken>  Percy ReadOnly or FullAccess Token
+  --day                       Generate today\'s summary report
```
Comment (nithyamn) [inline README.md:65]: Add this as "default"

File: src/summary.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        ...summary,
+        browsers,
+        projectName,
+        startDate,
+        endDate
+    }
+    if(!fs.existsSync('Summary') ){
+        fs.mkdirSync('Summary',{recursive:true})
+    }
+    fs.writeFileSync(`Summary/${summary.projectName}-${Date.now()}.json`,JSON.stringify(summary,undefined,2))
```
Comment (nithyamn) [inline src/summary.js:110]: Cosmetic: Can we add "week" or "prefix" to the report so its easier to distinguish. Or maybe group it in a folder.

File: README.md (hunk 3 above)
Comment (nithyamn) [reply README.md]: this needs to be incorporated

**Review by nithyamn — APPROVED** [review]

**Comment by nithyamn** [comment]
All reviews have been incorporated.

**Pull request merged**